src/services/role/model.js

```diff
@@ -20,53 +20,22 @@ const rolesDisplayName = {
 const roleSchema = new Schema({
 	name: { type: String, required: true },
 	permissions: [{ type: String }],
-
-	// inheritance
 	roles: [{ type: Schema.Types.ObjectId }],
 }, {
 	timestamps: true,
 });
 
-roleSchema.methods.getPermissions = function getPermissions() {
-	return roleModel.resolvePermissions([this._id]); // fixme
-};
-
-roleSchema.statics.resolvePermissions = function resolvePermissions(roleIds) {
-	const processedRoleIds = [];
-	const permissions = new Set();
-
-	function resolveSubRoles(roleId) {
-		return roleModel.findById(roleId) // fixme
-			.then((role) => {
-				if (typeof role !== 'object') {
-					role = {};
-				}
-				if (Array.isArray(role.permissions) === false) {
-					role.permissions = [];
-				}
-				role.permissions.forEach((p) => permissions.add(p));
-				const promises = role.roles
-					.filter((id) => !processedRoleIds.includes(id))
-					.map((id) => {
-						processedRoleIds.push(id);
-						return resolveSubRoles(id); // recursion
-					});
-				return Promise.all(promises);
-			});
-	}
-
-	return Promise.all(roleIds.map((id) => resolveSubRoles(id)))
-		.then(() => permissions);
-};
-
-roleSchema.virtual('displayName').get(function get() {
+/* roleSchema.virtual('displayName').get(function get() {
 	return rolesDisplayName[this.name] || '';
 });
-
-roleSchema.plugin(leanVirtuals);
+*/
+//roleSchema.plugin(leanVirtuals);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected exception block, space or tab after '//' in comment.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5229889)